pkg/target/clusters.go

```diff
@@ -40,29 +40,20 @@ func (cls Clusters) Send(finish chan struct{}) chan struct{} {
 func NewClusters(mainCfg conf.Main, cfg conf.Clusters, lg *zap.Logger, ms *metrics.Prom) (Clusters, error) {
 	// TODO (grzkv): Add duplicate defense
 	cls := make(map[string]*Cluster)
+	var err error
 	for _, cc := range cfg.Cluster {
 		cl := Cluster{
-			Name: cc.Name,
+			Name:                   cc.Name,
+			UpdateHostHealthStatus: make(chan *HostStatus),
+			Type:                   cc.Type,
 		}
 		switch cc.Type {
 		case conf.JumpCluster:
-			cl.Hosts = make([]*Host, len(cc.Hosts))
+			cl, err = getJumpCluster(cl, cc, mainCfg, lg, ms)
+		case conf.LB:
 			for _, h := range cc.Hosts {
-				if cl.Hosts[h.Index] != nil {
-					return cls, fmt.Errorf("duplicate index value, or index not set for a hashed cluster %s", cc.Name)
-				}
-
-				if h.Index >= len(cl.Hosts) {
-					return cls, fmt.Errorf("host %s index %d out of bounds in cluster %s (cluster size %d)", h.Name, h.Index, cl.Name, len(cl.Hosts))
-				}
-
-				cl.Hosts[h.Index] = NewHost(cc.Name, mainCfg, h, lg, ms)
-			}
-
-			for _, hst := range cl.Hosts {
-				if hst == nil {
-					return cls, fmt.Errorf("not all host indexes were set for cluster %v", cc.Name)
-				}
+				host := NewHost(cc.Name, mainCfg, h, lg, ms)
+				cl.Hosts = append(cl.Hosts, host)
```

Comment:
This seems to be identical with next switch case. How about just adding conf.LB to the list below?

Reply:
Indeed, reason being that clusters variable is properly defined before we try updating hosts status. Otherwise, the cluster itself did not exist due to the adding of cluster key at the end of the previous switch.